code/dqe/savedata.q

```diff
@@ -0,0 +1,18 @@
+\d .dqe
+savedata:{[dir;pt;tabname]                                                                                       /- function to save data persistently
+  .lg.o[`dqe;"Saving ",(string tabname)," data to ", string dir];
+  .[
+    upsert;
+    (` sv .Q.par[dir;pt;tabname],`;.Q.en[dir;r:0!.save.manipulate[tabname;`. tabname]]);                         /- upsert data to partition
+    {[e] .lg.e[`savedata;"Failed to save dqe data to disk : ",e];`e}
+  ];
```

Comment:
I'm really not a fan of this whole section, I would probably go for something like:
```
pth:` sv .Q.par[dir;pt;tabname],`;
err:{[e].lg.e[`savedata;"Failed to save dqe data to disk : ",e];'e}
.[upsert;(pth;.Q.en[dir;r:0!.save.manipulate[tabname;`. tabname]]);err];
```